vulnerabilities/tests/test_istio.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/vulnerablecode/
+# The VulnerableCode software is licensed under the Apache License version 2.0.
+# Data generated with VulnerableCode require an acknowledgment.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with VulnerableCode or any VulnerableCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnerableCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnerableCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnerableCode is a free software tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+
+import os
+from collections import OrderedDict
+from unittest import TestCase
+
+from packageurl import PackageURL
+
+from vulnerabilities.data_source import Advisory, Reference
+from vulnerabilities.importers.istio import IstioDataSource
+from vulnerabilities.package_managers import GitHubTagsAPI
+
+BASE_DIR = os.path.dirname(os.path.abspath(__file__))
+
+
+class TestIstioDataSource(TestCase):
+    @classmethod
+    def setUpClass(cls):
```

Comment:
Do you really need a setup when you have exactly one test function?
It creates IMHO un-necessary coupling between all the future unit tests and this is not meant to be shared for now in anycase.

Also you likely want to decouple the testing of the low level parsing of the .md with yaml frontmatter and the part that creates proper advisories from that

Reply:
Ok